### ISSUE TITLE
react/prop-types is not working when no jsx is used

### ISSUE THREAD

**Issue by TrejGun** [yannickcr/eslint-plugin-react#2305]
```js
import React from "react";
import PropTypes from "prop-types";

function FuncComponent({children, unUsedProp}) { // does not show error
  React.useEffect(() => {}, []);
  return children;
}

FuncComponent.propTypes = {
  children: PropTypes.node,
};
```

```js
import React from "react";
import PropTypes from "prop-types";

function FuncComponent({children, unUsedProp}) { // shows error
  React.useEffect(() => {}, []);
  return (
    <div>
      {children}
    </div>
  );
}

FuncComponent.propTypes = {
  children: PropTypes.node,
};
```

**Comment by ljharb** [comment]
If it's a function (not a class), and you're not using jsx, there's no way for static analysis to know it's a React component.

You could always `return <>children</>`, to avoid the need for the div.

**Comment by TrejGun** [comment]
what do you mean "there is no way"? 
there is definetly a way - MyFunction has `propTypes` prop 

yeah it may be difficult because of reassignment but still possible

```
const MyRefFunc = forwardRef(function FuncComponent({}, ref){})
MyRefFunc.propTypes = {};
```

**Comment by ljharb** [comment]
Yes, but the issue is with `FuncComponent`, not `MyRefFunc`, and `forwardRef` doesn't take a component - it takes a function callback. So, in this case, it's *not* a component.

**Comment by TrejGun** [comment]
yeah i did not check last example, apperently it works.

**Comment by ljharb** [comment]
I’d suggest wrapping children in a fragment (or, React.Children.only) to make it clear it’s a react component.
